web/channels/retro_channel.ex

```diff
@@ -117,6 +109,29 @@ defmodule RemoteRetro.RetroChannel do
     {:noreply, socket}
   end
 
+  defp add_idea!(%{"body" => body, "category" => "action-item", "userId" => user_id, "assigneeId" => assignee_id}, socket) do
+    %Idea{
```

Comment:
hmmm... there's still a lot of duplication between these two definitions. The only difference seems to be the `category` hardcoding and the addition of an `assignee_id`. I'm curious if, from the the front, you passed `assignee_id` as `null`, you could just use just the one definition in all cases. The reason i suspect that might work is that the default `assignee_id` value in the db is `NULL` anyway. @tnewell5 Can you give that a shot?

Reply:
about to tackle this bit - think it's the only one remaining

Reply:
@tnewell5 Since we're combining the two different cases into one, we don't actually need this `add_idea!` function. We can move the pattern matching and the Idea creation into the matching `handle_in` function clause.